File: app/src/main/java/net/ducksmanager/persistence/models/dm/Purchase.java
package net.ducksmanager.persistence.models.dm;

import com.google.gson.annotations.Expose;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "purchases")
public class Purchase {
    @Expose
    @PrimaryKey
    @ColumnInfo(name = "purchaseId")
    private Integer id;

    @Expose
    @ColumnInfo
    private String date;

    @Expose
    @ColumnInfo
    private String description;

    @Ignore
    public Purchase() {}

    public Purchase(String date, String description) {
        this.date = date;
        this.description = description;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getDate() {
        return date;
    }

    public String getDescription() {
        return description;
    }

}

File: app/src/main/java/net/ducksmanager/persistence/models/dm/Purchase.kt
package net.ducksmanager.persistence.models.dm

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.google.gson.annotations.Expose

@Entity(tableName = "purchases")
open class Purchase {
    @Expose
    @PrimaryKey
    @ColumnInfo(name = "purchaseId")
    var id: Int? = null

    @Expose
    @ColumnInfo
    var date: String? = null
        private set

    @Expose
    @ColumnInfo
    var description: String? = null
        private set

    @Ignore
    constructor()

    constructor(date: String?, description: String?) {
        this.date = date
        this.description = description
    }
}